src/tools/bump-stage0/src/main.rs

```diff
@@ -4,7 +4,7 @@ use indexmap::IndexMap;
 use std::collections::HashMap;
 
 const PATH: &str = "src/stage0.json";
-const COMPILER_COMPONENTS: &[&str] = &["rustc", "rust-std", "cargo"];
```

Comment:
I thought we had dropped the `preview` from the clippy component somehow, or where is this coming from?

When I check in rustup I can see `clippy-x86_64-unknown-linux-gnu` but maybe `-preview` still works as a fallback?

Reply:
clippy is the alias, not the canonical name
```
$ curl https://static.rust-lang.org/dist/2023-11-04/channel-rust-nightly.toml | grep clippy
[renames.clippy]
to = "clippy-preview"
complete = ["rustc", "cargo", "rust-std", "rust-mingw", "rust-docs", "rustfmt-preview", "clippy-preview", "rls-preview", "rust-analyzer-preview", "rust-src", "llvm-tools-preview", "rust-analysis", "miri-preview", "rustc-codegen-cranelift-preview", "rustc-dev"]
default = ["rustc", "cargo", "rust-std", "rust-mingw", "rust-docs", "rustfmt-preview", "clippy-preview"]
$ ; tar -tf build/cache/2023-10-04/clippy-beta-x86_64-unknown-linux-gnu.tar.xz  | head -n6
clippy-beta-x86_64-unknown-linux-gnu
clippy-beta-x86_64-unknown-linux-gnu/clippy-preview
clippy-beta-x86_64-unknown-linux-gnu/clippy-preview/share
clippy-beta-x86_64-unknown-linux-gnu/clippy-preview/share/doc
clippy-beta-x86_64-unknown-linux-gnu/clippy-preview/share/doc/clippy
clippy-beta-x86_64-unknown-linux-gnu/clippy-preview/bin
```